From f85ae0efd949e3b7ca7c2c21d96bc8ca952fd52b Mon Sep 17 00:00:00 2001
From: "Hsia-Jun(Randy) Li" <randy.li@synaptics.com>
Date: Tue, 20 Aug 2024 17:55:07 +0800
Subject: [PATCH 09/20] v4l2bufferpool: actually queue back the empty buffer
 flagged LAST

Fixes
9b88ebe05b997af0183474a7f8c63530f9ac4b27
The buffer would fail at gst_v4l2_is_buffer_valid() before,
since it has a reference on it, it is not writable.

Signed-off-by: Hsia-Jun(Randy) Li <randy.li@synaptics.com>
---
 subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
index 38569bff5d..0dcc5bad5f 100644
--- a/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
+++ b/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c
@@ -1293,10 +1293,8 @@ gst_v4l2_buffer_pool_dqbuf (GstV4l2BufferPool * pool, GstBuffer ** buffer,
   if (group->buffer.flags & V4L2_BUF_FLAG_LAST &&
       group->planes[0].bytesused == 0) {
     GST_DEBUG_OBJECT (pool, "Empty last buffer, signalling eos.");
-    *buffer = outbuf;
-    outbuf = NULL;
-    gst_buffer_ref (*buffer);
-    gst_v4l2_buffer_pool_complete_release_buffer (bpool, *buffer, FALSE);
+    gst_v4l2_buffer_pool_complete_release_buffer (bpool, outbuf, FALSE);
+    *buffer = NULL;
     goto eos;
   }
 
-- 
2.17.1

